src/lsm/scan_lookup.zig

```diff
@@ -23,6 +23,14 @@ pub fn ScanLookupType(
 
         pub const Callback = *const fn (*ScanLookup) void;
 
+        pub const Status = enum {
+            idle,
+            scan,
+            lookup,
+            buffer_finished,
+            scan_finished,
```

Comment:
When a scan finishes, it was because there are no more results (`scan_finished`) or because the entire buffer was filled (`buffer_finished`).

In this case, we need to know if we still need to process the same expiry timestamp again in the next pulse (e.g., there are more than 8190 transfers to expire).

In general, the QueryAPI will need this to indicate the pagination status.